src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java

```diff
@@ -1,65 +1,37 @@
 package edu.kis.vh.nursery;
 
-public class DefaultCountingOutRhymer {
-
-	private static final int ERROR_VALUE = -1;
-
-	private static final int NUMBERS_SIZE = 12;
+import edu.kis.vh.nursery.structures.IntArrayStack;
+import edu.kis.vh.nursery.structures.IntLinkedList;
+import edu.kis.vh.nursery.structures.StructuresInterface;
 
-	private final int[] numbers = new int[NUMBERS_SIZE];
+public class DefaultCountingOutRhymer {
+	private final StructuresInterface stack;
 
-	private int total = -1;
+	public DefaultCountingOutRhymer() {
+		stack = new IntLinkedList();
+	}
 
-	/**
-	 * Get current size of stack
-	 * @return size
-	 */
-	public int getTotal() {
-		return total;
+	public DefaultCountingOutRhymer(IntArrayStack intArrayStack){
```

Comment:
The constructor should take StructInterface as an argument. It would be the most consistent and flexible approach.